4/Dockerfile

```diff
@@ -19,10 +19,22 @@ WORKDIR /var/www/html
 RUN	apk add --no-cache libpq
 
 RUN	set -x \
+&&	apk add --no-cache --virtual .php-ext-deps \
+	unixodbc \
+	freetds \
```

Comment:
Can you check whether something like the following from Wordpress works: https://github.com/docker-library/wordpress/blob/0df5de06a4f43f2790dfc3be92554a7e229115d9/Dockerfile-alpine.template#L36-L42

It should figure out the runtime dependencies itself then.

Reply:
Gave this a shot but it didn't work.  Note I am pretty clueless about this build chain in general, the image builds, but is 400mb in size,  mssql doesn't work, and the following is seen during build;

```
> + tr , '\n'
> + awk 'system("[ -e /usr/local/lib/" $1 " ]") == 0 { next } { print "so:" $1 }'
> + sort -u
> + scanelf --needed --nobanner --format '%n#p' --recursive /usr/local/lib/php/extensions
> + docker-php-ext-install pdo_mysql pdo_pgsql pdo_sqlite pdo_odbc pdo_dblib 'runDeps=so:libc.musl-x86_64.so.1
> so:libsodium.so.23'
> error: /usr/src/php/ext/runDeps=so:libc.musl-x86_64.so.1
> so:libsodium.so.23 does not exist
> 
> usage: /usr/local/bin/docker-php-ext-install [-jN] ext-name [ext-name ...]
>    ie: /usr/local/bin/docker-php-ext-install gd mysqli
>        /usr/local/bin/docker-php-ext-install pdo pdo_mysql
>        /usr/local/bin/docker-php-ext-install -j5 gd mbstring mysqli pdo pdo_mysql shmop
> 
> if custom ./configure arguments are necessary, see docker-php-ext-configure
> 
> Possible values for ext-name:
> bcmath bz2 calendar ctype curl dba dom enchant exif ffi fileinfo filter ftp gd gettext gmp hash iconv imap intl json ldap mbstring mysqli oci8 odbc opcache pcntl pdo pdo_dblib pdo_firebird pdo_mysql pdo_oci pdo_odbc pdo_pgsql pdo_sqlite pgsql phar posix pspell readline reflection session shmop simplexml snmp soap sockets sodium spl standard sysvmsg sysvsem sysvshm tidy tokenizer xml xmlreader xmlrpc xmlwriter xsl zend_test zip
> 
> Some of the above modules are already compiled into PHP; please check
> the output of "php -i" to see which modules are already loaded.
> + apk add --virtual .phpexts-rundeps
``` 

https://github.com/TimWolla/docker-adminer/compare/master...staff0rd:scanelf?expand=1

Reply:
@staff0rd The issue is that you did not start a new shell command. In Wordpress this is done by the semicolon after `docker-php-ext-enable imagick`. The adminer image is still using the “old style” `&&` to chain the commands instead of setting `set -e` and using a semicolon.

I have just checked locally. The following appears to build properly:
```
&&	docker-php-ext-install \
	pdo_mysql \
	pdo_pgsql \
	pdo_sqlite \
	pdo_odbc \
	pdo_dblib \
&&	runDeps="$( \
		scanelf --needed --nobanner --format '%n#p' --recursive /usr/local/lib/php/extensions \
			| tr ',' '\n' \
			| sort -u \
			| awk 'system("[ -e /usr/local/lib/" $1 " ]") == 0 { next } { print "so:" $1 }' \
	)" \
&&	apk add --virtual .phpexts-rundeps $runDeps \
&&	apk del .build-deps
```